tests/models/mock_async_stream.py

```diff
@@ -47,7 +47,10 @@ async def __anext__(self) -> T:
         synchronous iterator. If the iterator is exhausted, `StopAsyncIteration`
         is raised.
         """
-        return _utils.sync_anext(self._iter)
+        next = _utils.sync_anext(self._iter)
+        if isinstance(next, Exception):
+            raise next
```

Comment:
```suggestion
        raise_if_exception(next)
```
I mean you have the function lol